Automattic/jetpack
# Instant Search: Polish search interface UX

Please note that the commit log includes previously merged changes for Instant Search. The new code introduced by this PR is from Feb 12, 2020, and onwards.

#### Changes proposed in this Pull Request:
* This PR primarily focuses on improving the overall user experience for Jetpack Instant Search.

>Before:
>
><img width="1035" alt="Screen Shot 2020-02-21 at 3 42 30 PM" src="https://user-images.githubusercontent.com/4044428/75077724-d7d90700-54c0-11ea-83d0-f8f8f41df3e4.png">

>After:
>
><img width="1035" alt="Screen Shot 2020-02-21 at 3 40 31 PM" src="https://user-images.githubusercontent.com/4044428/75077732-dc9dbb00-54c0-11ea-9b90-3e6f81563248.png">

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* No, this is an incremental update for Jetpack Instant Search.

#### Testing instructions:
1. Follow [these setup instructions](https://github.com/Automattic/jetpack/blob/instant-search-master/modules/search/instant-search/README.md#testing-instructions) to set up Jetpack Instant Search on your site. Please note that the Jetpack Search widget will need to be added to twice: once to your site sidebar/footer and once to the Jetpack Search Sidebar within the search overlay.

2. Submit a query into your site search input. Ensure that a search overlay appears with your search results.

3. Try using the new sort select controls below the overlay search input. Ensure that they behave as expected.

4. Try resizing your window to be narrower than 768 pixels; a Filters button should spawn adjacent to the sort select controls.

5. Try clicking on the Filters button in (4). Ensure that a filters popover spawns below the button, like so:
<img width="388" alt="Screen Shot 2020-02-21 at 3 47 34 PM" src="https://user-images.githubusercontent.com/4044428/75077958-7f563980-54c1-11ea-8300-e7d6c9036ba0.png">

#### Proposed changelog entry for your changes:
* Interface refinements and bugfixes for Jetpack Instant Search.
